### ISSUE TITLE
[nrk:series] No video formats found / 404 not found

### ISSUE THREAD

**Issue by YumekuiNeru** [ytdl-org/youtube-dl#24221]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.03.01**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
$ youtube-dl 'https://tv.nrk.no/serie/harald-eia-presenterer-saann-er-norge/sesong/1/episode/1/avspiller' --verbose
[debug] System config: []
[debug] User config: ['--buffer-size', '16K']
[debug] Custom config: []
[debug] Command-line args: ['https://tv.nrk.no/serie/harald-eia-presenterer-saann-er-norge/sesong/1/episode/1/avspiller', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.03.01
[debug] Python version 3.6.10 (CPython) - Linux-5.5.5-gentoo-pix-x86_64-Intel-R-_Core-TM-_i7-6700K_CPU_@_4.00GHz-with-gentoo-2.7
[debug] exe versions: ffmpeg 4.2.2, ffprobe 4.2.2, rtmpdump 2.4
[debug] Proxy map: {}
[NRKTVEpisode] harald-eia-presenterer-saann-er-norge/sesong/1/episode/1: Downloading webpage
[debug] Using fake IP 84.212.51.255 (NO) as X-Forwarded-For.
[NRK] KMTE31000118: Downloading mediaelement JSON
[NRK] KMTE31000118: Downloading f4m manifest
WARNING: Unable to download f4m manifest: HTTP Error 404: Not Found
[NRK] KMTE31000118: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[NRK] KMTE31000118: Downloading f4m manifest
WARNING: Unable to download f4m manifest: HTTP Error 404: Not Found
[NRK] KMTE31000118: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib64/python3.6/site-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib64/python3.6/site-packages/youtube_dl/extractor/nrk.py", line 96, in _real_extract
    self._sort_formats(formats)
  File "/usr/lib64/python3.6/site-packages/youtube_dl/extractor/common.py", line 1327, in _sort_formats
    raise ExtractorError('No video formats found')
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.


```

## Description

I see mentions of a fargerik-psapi.nrk.no in some dumped files.

**Comment by Krissz** [comment]
It's a same error :(

```
$ youtube-dl https://www.rtlmost.hu/draga-orokosok-p_12633/draga-orokosok-3-evad-43-resz-c_12590651 --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://www.rtlmost.hu/draga-orokosok-p_12633/draga-orokosok-3-evad-43-resz-c_12590651', u'--verbose']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.03.01
[debug] Python version 2.7.12 (CPython) - Linux-4.4.59+-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 2.7.1
[debug] Proxy map: {}
[6play] 12590651: Downloading JSON metadata
[6play] 12590651: Downloading webpage
[6play] 12590651: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[6play] 12590651: Downloading MPD manifest
WARNING: Failed to download MPD manifest: HTTP Error 404: Not Found
[6play] 12590651: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[6play] 12590651: Downloading MPD manifest
WARNING: Failed to download MPD manifest: HTTP Error 404: Not Found
[6play] 12590651: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[6play] 12590651: Downloading MPD manifest
WARNING: Failed to download MPD manifest: HTTP Error 404: Not Found
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/sixplay.py", line 113, in _real_extract
    self._sort_formats(formats)
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/common.py", line 1327, in _sort_formats
    raise ExtractorError('No video formats found')
ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

**Comment by Magixa** [comment]
I have the same problem with videos posted in March on rtlmost.hu. I can still download everyting that was posted in February and before that.

**Comment by YumekuiNeru** [comment]
> I have the same problem with videos posted in March on rtlmost.hu. I can still download everyting that was posted in February and before that.

that uses a different extractor than the nrk one, though?

**Comment by Roxedus** [comment]
Spent a few days looking into it, for me it looks like Akamai now offers a new way to serve manifests, which common._extract_akamai_formats cannot find (I had no luck either with Developer console in Chrome). https://github.com/ytdl-org/youtube-dl/blob/7947a1f7dbc6ba47a6f22ab67fd330e57c0ef87c/youtube_dl/extractor/common.py#L2562-L2583

This show is the only one I found on NRK that behaves like this, I have checked other shows with similar traits, like geo-block (this one shouldn't be), and same production year (2020).

fargerik-psapi.nrk.no is only used to personalize the site, looking at the indexed docs for the api.

**Comment by YumekuiNeru** [comment]
If it is useful, including the part that specifies the episode in the URL makes the download work for episodes 3-8, but fails for 1+2

**Comment by Roxedus** [comment]
It didn't exactly help me, *but* I did manage to get it to download those two episodes. However I cant find any value in the api to differentiate the working ones, vs the "broken" ones.

It looks like NRK did something wrong on the two first episodes, which also can be confirmed when looking at the data object created in nrk.py#34, and the `data.mediaAssetsOnDemand.0.hlsUrl` value which returns the same url as the one `common._extract_akamai_formats` function sends to `_extract_m3u8_formats`, which does not work for the two episodes, but the others.


The URLs from the CDN looks to be split on availability, worldwide and Norway, looking at the start of the url.

`http://*.akamaihd.net/i/no/<...>/master.m3u8` - Norway

`http://*.akamaihd.net/i/wo/<...>/master.m3u8` - Worldwide


Adding a regular expression to replace wo to no, fixes it for these two episodes, but breaks anything that's available worldwide, in other words, the rest of the season. I have not found any indicator to determine if this replacement is necessary before attempting to download. Any "fix" by me will be crude, and depend on a try/expect kinda deal. 

To get the two episodes I added a new line after line 62 in nrk.py.
`asset_url = re.sub(r'(https?://[^/]+)/z/wo/', r'\1/z/no/', asset_url)`

Either this is a external-bugs or a wont-fix, unless someone well more versed in the project can figure out a more elegant way.
